numpy/_core/src/multiarray/descriptor.c

```diff
@@ -3093,6 +3104,10 @@ arraydescr_setstate(PyArray_Descr *self, PyObject *args)
      * flags as an int even though it actually was a char in the PyArray_Descr
      * structure
      */
+    if (int_dtypeflags < 0 && int_dtypeflags >= -128) {
```

Comment:
Should one error on `int_dtypeflags < -128`?

Reply:
It'll error below...

Reply:
Ah, of course!